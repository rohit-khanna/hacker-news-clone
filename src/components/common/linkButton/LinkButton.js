import React from "react";
import "./LinkButton.scss";

export default function LinkButton({
  children = "button",
  title,
  className,
  onClick,
  ...props
}) {
  return (
    <button
      {...props}
      className={`linkButton cursor-pointer ${className ? className : ""}`}
      title={title}
      onClick={onClick}
    >
      {children}
    </button>
  );
}
