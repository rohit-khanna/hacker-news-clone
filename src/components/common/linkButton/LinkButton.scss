@import "src/resources/css/variables.scss";

.linkButton {
  background-color: transparent;
  color: $primaryColor;
  border: none;
  font-weight: normal;
  text-transform: none;
  min-width: 0;
  padding: 0.15rem 0.25rem;
  outline: none;
  text-transform: capitalize;
  font-weight: bold;
  letter-spacing: 1px;
  &:hover {
    text-decoration: underline;
  }
  &:active,
  &:focus {
    outline: none;
  }
}
