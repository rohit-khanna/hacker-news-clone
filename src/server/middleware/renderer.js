import React from "react";
import ReactDOMServer from "react-dom/server";
import { StaticRouter, Switch, Route } from "react-router-dom";
// import our main App component
import App from "../../App";
import { Provider as ReduxProvider } from "react-redux";
import ConfigureStore from "./configureStore";

const AppRoutes = () => {
  return (
    <Switch>
      <Route path="/" exact component={App} />
    </Switch>
  );
};

export default (req, res) => {
  console.log("adasdasd");
  const store = new ConfigureStore();
  const html = ReactDOMServer.renderToString(
    <StaticRouter context={{}}>
      {/* <ReduxProvider store={store}> */}
      <AppRoutes />
      {/* </ReduxProvider> */}
    </StaticRouter>
  );

  console.log(html);

  res.send(`<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="theme-color" content="#000000" />
    <title>Hacker News Clone</title>
  </head>
  <body>  <noscript>You need to enable JavaScript to run this app.</noscript>
    <div id="root">${html}</div>
    <script>
      window.__PRELOADED_STATE__ = ${JSON.stringify(store.getState())}
    </script>
    <script src="bundle.js"></script>
  </body>
</html>
`);

  // point to the html file created by CRA's build tool
  // const filePath = path.resolve(__dirname, "..", "build", "index.html");
  //   fs.readFile(filePath, "utf8", (err, htmlData) => {
  //     if (err) {
  //       console.error("err", err);
  //       return res.status(404).end();
  //     }
  //     const store = new ConfigureStore();
  //     /// We can make a Call to fetch Data from HERE >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
  //     // TODO
  //     // render the app as a string
  //     const html = ReactDOMServer.renderToString(
  //       <StaticRouter location={req.url} context={{}}>
  //         <ReduxProvider store={store}>
  //           <AppContainer />
  //         </ReduxProvider>
  //       </StaticRouter>
  //     );
  //     // inject the rendered app into our html and send it
  //     res.send(
  //       htmlData.replace(
  //         '<div id="root"></div>',
  //         `
  //       <div id="root">${html}</div>
  //       <script>
  //         window.__PRELOADED_STATE__ = ${JSON.stringify(store.getState())}
  //       </script>
  // `
  //       )
  //     );
  //   });
};
