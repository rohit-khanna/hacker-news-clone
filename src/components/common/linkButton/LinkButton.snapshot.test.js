import renderer from "react-test-renderer";
import React from "react";
import LinkButton from "./index";

describe("<LinkButton/>", () => {
  it("should match the snapshot", () => {
    const tree = renderer.create(
      <LinkButton className="someClass" title="Some title" onClick={() => {}}>
        some text
      </LinkButton>
    );
    expect(tree).toMatchSnapshot();
  });
});
