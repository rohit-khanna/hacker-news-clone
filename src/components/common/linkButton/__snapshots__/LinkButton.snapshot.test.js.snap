// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton/> should match the snapshot 1`] = `
<button
  className="linkButton cursor-pointer someClass"
  onClick={[Function]}
  title="Some title"
>
  some text
</button>
`;
